### ISSUE TITLE
Required pre-requisite test is not executed when it's not part of the filete

### ISSUE THREAD

**Issue by jefersonralmeida** [Codeception/Codeception#5427]
#### What are you trying to achieve?
I have those tests scenarios inside a Cest:
```php
<?php
    /**
     * @param ApiTester $I
     */
    public function testA(ApiTester $I)
    {
        // my assertions
    }

    /**
     * @param ApiTester $I
     * @group b
     * @depends testA
     */
    public function testB(ApiTester $I)
    {
        // my assertions
    }
```
Executing `codecept run --group=b` I would like to have a way to force testA to be executed, even if it is not part of the filter, since it's a requirement for testB. Maybe we could have a config option like `forcePreRequisiteExecution` on the YML file.

This would be really useful.

#### What do you get instead?

`testB` is skipped, since testA wasn't executed;

Running `codecept run --group=b -vvv` I get:

```bash
Codeception PHP Testing Framework v2.5.4
Powered by PHPUnit 7.5.6 by Sebastian Bergmann and contributors.
Running with seed:

[Groups] b

Api Tests (1) -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Modules: REST, PhpBrowser, \Helper\Api
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
LoginAdminCest: Test b
Signature: LoginAdminCest:testB
Test: tests\login\LoginAdminCest.php:testB
Scenario --
 SKIPPED: This test depends on LoginAdminCest:testA to pass
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


Time: 1.19 seconds, Memory: 12.00 MB

There was 1 skipped test:

---------
1) LoginAdminCest: Test b
 Test  tests\login\LoginAdminCest.php:testB
This test depends on LoginAdminCest:testA to pass

OK, but incomplete, skipped, or risky tests!
Tests: 1, Assertions: 0, Skipped: 1.
```

### Details

* Codeception version: 2.5.4
* PHP Version: 7.2.13
* Operating System: Windows 10
* Installation type: composer
* List of installed packages:
```
behat/gherkin                      v4.6.0  Gherkin DSL parser for PHP 5.3
codeception/codeception            2.5.4   BDD-style testing framework
codeception/phpunit-wrapper        7.7.1   PHPUnit classes used by Codeception
codeception/stub                   2.1.0   Flexible Stub wrapper for PHPUnit's Mock Builder
doctrine/instantiator              1.1.0   A small, lightweight utility to instantiate objects in PHP without invoking their constructors
facebook/webdriver                 1.6.0   A PHP client for Selenium WebDriver
guzzlehttp/guzzle                  6.3.3   Guzzle is a PHP HTTP client library
guzzlehttp/promises                v1.3.1  Guzzle promises library
guzzlehttp/psr7                    1.5.2   PSR-7 message implementation that also provides common utility methods
mongodb/mongodb                    1.4.2   MongoDB driver library
myclabs/deep-copy                  1.8.1   Create deep copies (clones) of your objects
phar-io/manifest                   1.0.3   Component for reading phar.io manifest information from a PHP Archive (PHAR)
phar-io/version                    2.0.1   Library for handling version information and constraints
phpdocumentor/reflection-common    1.0.1   Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock  4.3.0   With this component, a library can provide support for annotations via DocBlocks or otherwise retrieve information that is embedded in a DocBlock.
phpdocumentor/type-resolver        0.4.0
phpoption/phpoption                1.5.0   Option Type for PHP
phpspec/prophecy                   1.8.0   Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage          6.1.4   Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator          2.0.2   FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template          1.2.1   Simple template engine.
phpunit/php-timer                  2.1.1   Utility class for timing
phpunit/php-token-stream           3.0.1   Wrapper around PHP's tokenizer extension.
phpunit/phpunit                    7.5.6   The PHP Unit Testing framework.
psr/http-message                   1.0.1   Common interface for HTTP messages
ralouphie/getallheaders            2.0.5   A polyfill for getallheaders.
sebastian/code-unit-reverse-lookup 1.0.1   Looks up which function or method a line of code belongs to
sebastian/comparator               3.0.2   Provides the functionality to compare PHP values for equality
sebastian/diff                     3.0.2   Diff implementation
sebastian/environment              4.1.0   Provides functionality to handle HHVM/PHP environments
sebastian/exporter                 3.1.0   Provides the functionality to export PHP variables for visualization
sebastian/global-state             2.0.0   Snapshotting of global state
sebastian/object-enumerator        3.0.3   Traverses array structures and object graphs to enumerate all referenced objects
sebastian/object-reflector         1.1.1   Allows reflection of object attributes, including inherited and non-public ones
sebastian/recursion-context        3.0.0   Provides functionality to recursively process PHP variables
sebastian/resource-operations      2.0.1   Provides a list of PHP built-in functions that operate on resources
sebastian/version                  2.0.1   Library that helps with managing the version number of Git-hosted PHP projects
symfony/browser-kit                v4.2.4  Symfony BrowserKit Component
symfony/console                    v4.2.4  Symfony Console Component
symfony/contracts                  v1.0.2  A set of abstractions extracted out of the Symfony components
symfony/css-selector               v4.2.4  Symfony CssSelector Component
symfony/dom-crawler                v4.2.4  Symfony DomCrawler Component
symfony/event-dispatcher           v4.2.4  Symfony EventDispatcher Component
symfony/finder                     v4.2.4  Symfony Finder Component
symfony/polyfill-ctype             v1.10.0 Symfony polyfill for ctype functions
symfony/polyfill-mbstring          v1.10.0 Symfony polyfill for the Mbstring extension
symfony/process                    v4.2.4  Symfony Process Component
symfony/yaml                       v4.2.4  Symfony Yaml Component
theseer/tokenizer                  1.1.0   A small library for converting tokenized PHP source code into XML and potentially other formats
vlucas/phpdotenv                   v3.3.2  Loads environment variables from `.env` to `getenv()`, `$_ENV` and `$_SERVER` automagically.
webmozart/assert                   1.4.0   Assertions to validate method input/output with nice error messages.
```
* Suite configuration:

```yml
# suite config
suites:
    api:
        actor: ApiTester
        path: .
        modules:
            enabled:
                - REST:
                    url: http://localhost
                    depends: PhpBrowser
                - Helper\Api

paths:
    tests: tests
    output: tests/_output
    data: tests/_data
    support: tests/_support

settings:
    shuffle: false
    lint: true
    report_useless_tests: true

coverage:
    enabled: true
    remote: true
    remote_context_options:
        http:
            timeout: 10800
```

Ps.: I'm planning to create a PR to implement this feature, but I want to bring it to discussion before starting to code.

**Comment by piotroxp** [comment]
This would be really useful for testing intricate scenarios, and gives an added amount of flexibility

**Comment by Naktibalda** [comment]
You are welcome to implement it.
